### PR TITLE
Add MCO Bootstrap E2E test job

### PR DESCRIPTION
**Pull request by JoelSpeed**
This adds a test job to run the new `bootstrap-e2e` test being add to MCO in the following PR
- https://github.com/openshift/machine-config-operator/pull/2687
Once this PR is merged, there will be an envtest based test suite that verifies that day 1 and day 2 rendered configs are consistent, to prevent errors when MCD cannot find the config it was bootstrapped with (more details on the PR).

This also removes the e2e-techpreview job which has been in place as an interim to make sure that a featuregate being added on install did not cause the cluster to degrade because of discrepancies in MCOs day 1 and day 2 config.

CC @yuqi-zhang @sinnykumari

### CODE REVIEW

**Comment by JoelSpeed** [comment]
We may wish to make this optional at first? Any thoughts on that from MCO team?

File: ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-master.yaml [hunk 1]
```diff
@@ -80,16 +80,16 @@ tests:
...
-  steps:
-    cluster_profile: aws
-    workflow: openshift-e2e-aws-techpreview
 - as: verify
   commands: |
     make verify
   container:
     from: src
+- as: bootstrap-e2e
+  optional: true
```
Comment (sinnykumari) [inline ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-master.yaml:89]: test is failing. Looks like ci-operator-config expects values in alphabetical order. Also, I am wondering if you are fine with moving optional field into presubmits file to maintain consistency.

Comment (JoelSpeed) [reply ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-master.yaml]: > Also, I am wondering if you are fine with moving optional field into presubmits file to maintain consistency.

It was my understanding that the optional field was moved out to the ci-config to make it more explicit? There are other examples of `optional` being in the ci-config within this file (most seem to have it here)

Comment (sinnykumari) [reply ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-master.yaml]: you are right, after looking carefully I do see it has been moved to ci-config. Looks like I missed when this change was made.

**Pull request merged**